### DISCUSSION TITLE
How to add a custom symfony form?

### DISCUSSION THREAD

**Discussion by nlipkanov** [sulu/SuluFormBundle#339] [Q&A] [answered]
### Custom Symfony Form & Twig Templates
My goal is to create a pure symfony form in sulu. I am a using the default sulu admin ui and have not extended anything.
I have a contact page (a twig html page) where I want to render my custom symfony form into.

I am not using any custom controllers or entites etc.

How do I do that?

**Comment by alexander-schranz** [comment, marked as answer]
@nlipkanov for "pure symfony" form you don't need a SuluFormBundle you [Create a Custom Controller for your Sulu Template](https://docs.sulu.io/en/2.5/cookbook/custom-controller.html) and handle the Form there like you would do it in a normal Symfony Application.

The Sulu Form Bundle allows the Content Manager to create Forms, this specially make sense for simple forms like a "Contact" Form. Any form without a special business Logic behind it. So the Content Manager can decide which fields he want and should be shown for that form.

**Reply by nlipkanov** [reply]
Okay, but how do I access e.g. like `$attributes['myData'] = 'data from frontend controller!';` in my twig template?

**Reply by alexander-schranz** [reply]
It is available under `myData` if you don't know your available variables do `{{ dump() }}` in twig.

**Reply by nlipkanov** [reply]
Thanks! I see my variable now.
Is this anywhere documented? Because I could not find it in the docs.
